### ISSUE TITLE
Show current status of notification access granted in Settings

### ISSUE THREAD

**Issue by ankitdbst** [saurabhere/BugMeLater#1]
Haven't found any solution to listen the notification access change to our app. One can detect if access is granted by checking in `NotificationCompat::getEnabledListenerPackages`.

**Comment by saurabhere** [comment]
you could run a loop on all active services and see if MYNotificationListenerService is one of them.
```
ContentResolver contentResolver = context.getContentResolver();
String enabledNotificationListeners = Settings.Secure.getString(contentResolver, "enabled_notification_listeners");
String packageName = context.getPackageName();

// check to see if the enabledNotificationListeners String contains our package name
if (enabledNotificationListeners == null || !enabledNotificationListeners.contains(packageName))
{
    // in this situation we know that the user has not granted the app the Notification access permission
    throw new Exception();
}
else
{
    doSomethingThatRequiresNotificationAccessPermission();
}
```

**Comment by ankitdbst** [comment]
Yeah, but there is no way to listen to the changes. Basically in the settings activity, if the User has revoked the notification access, then have no way to know that the preference has changed.

**Comment by saurabhere** [comment]
https://github.com/saurabhere/Renotify/commit/487ffb2b8df1f9442890c6403b1bf0a6d59179f5 
fixes this to some extent.
I have left scope for you to change the way the app looks in that case. Maybe show all settings blurred with a hovering pointer to the grant notifications access button.
Also, after addition of the splash screen we should move the grant permissions to the options menu as it is not very much needed after the initial usage.

**Comment by ankitdbst** [comment]
> 487ffb2
> fixes this to some extent.
> I have left scope for you to change the way the app looks in that case. Maybe show all settings blurred with a hovering pointer to the grant notifications access button.

I can do that, but it would only make sense for the first time. Subsequently, until I know the state of permission I cannot blur it.

> Also, after addition of the splash screen we should move the grant permissions to the options menu as it is not very much needed after the initial usage.

No, I think from usability point of view, these settings should be part of the Settings Activity because
- One click and always visible to the user
- It can show whether Renotify is active or not depending upon the permissions

**Comment by saurabhere** [comment]
Try running the updated app and you'll see that the permission check works
everytime you see the SettingsActivity. The magic is in the onResume()
method.

The app intro has provision to set permission from itself. So, the use case
for giving permissions is not left in the SettingsActivity. For status, we
can use blurring etc. Currently I am showing a toast.

On 23-Dec-2016 6:31 AM, "Ankit Gupta" <notifications@github.com> wrote:

487ffb2
<https://github.com/saurabhere/Renotify/commit/487ffb2b8df1f9442890c6403b1bf0a6d59179f5>
fixes this to some extent.
I have left scope for you to change the way the app looks in that case.
Maybe show all settings blurred with a hovering pointer to the grant
notifications access button.
I can do that. But it would only make sense for the first time.
Subsequently, until I know the state of permission I cannot blur it.

Also, after addition of the splash screen we should move the grant
permissions to the options menu as it is not very much needed after the
initial usage.
No, I think from usability point of view, these settings should be part of
the Settings Activity. Because


   - One click and always visible to the user
   - It can show whether Renotify is active or not depending upon the
   permissions

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/saurabhere/Renotify/issues/1#issuecomment-268923480>,
or mute the thread
